src/modules/file/file.usecase.ts

```diff
@@ -158,29 +160,51 @@ export class FileUseCases {
   async updateFileMetaData(
     user: User,
     fileUuid: File['uuid'],
-    newFileMetada: UpdateFileMetaDto,
+    newFileMetadata: UpdateFileMetaDto,
   ) {
+    if (
+      isStringEmpty(newFileMetadata.plainName) &&
+      isStringEmpty(newFileMetadata.type)
+    ) {
+      throw new BadRequestException('Filename cannot be empty');
+    }
+
     const file = await this.fileRepository.findOneBy({
       uuid: fileUuid,
       status: FileStatus.EXISTS,
+      userId: user.id,
     });
+    if (!file) {
+      throw new NotFoundException('File not found');
+    }
 
-    if (!file.isOwnedBy(user)) {
-      throw new ForbiddenException('This file is not yours');
+    let plainName =
+      newFileMetadata.plainName ??
+      file.plainName ??
+      this.cryptoService.decryptName(file.name, file.folderId);
+    if (isStringEmpty(plainName)) {
```

Comment:
Please, enforce these checks using the File entity rather than keeping all the business rules exposed here